### DISCUSSION TITLE
Question about Table.Column prop typescript

### DISCUSSION THREAD

**Discussion by wong2** [zeit-ui/react#746] [Q&A]
geist Table.Column component checks `prop` field be key of the passed data, but what if I need to render two columns by the same key? For example if there is a `user` struct in my data, I need to display `user.name` and `user.id` in different columns

**Comment by unix** [comment]
At the component level, there is no particularly good way to, However, you can use a simple data transformation to make the source data fit the requirements：

```js
const data = source.reduce((pre, next) => {
  if (!isUser(next)) return [...pre, next]
  const { name, id } = next
  return [...pre, { ...next, value: name }, { ...next, value: id }]
}, [])
```

After you do anything, you can still use `reduce` to assemble the data back.
